web/app.js

```diff
@@ -560,7 +560,10 @@ let PDFViewerApplication = {
   setTitleUsingUrl(url) {
     this.url = url;
     this.baseUrl = url.split('#')[0];
-    let title = getPDFFileNameFromURL(url, '');
+    let title = this.attachedFileName;
+    if (title === null) {
+      title = getPDFFileNameFromURL(url, '');
+    }
```

Comment:
Why not simply `let title = this.attachedFileName || getPDFFileNameFromURL(url, '');` instead?
That way, you'd also avoid issues with the title being empty if the file name found in the content disposition header is e.g. an empty string.